script.rb

```diff
@@ -0,0 +1,148 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield self[i]
+
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  # my_select
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    selected_array = []
+    i = 0
+    until i == size
+      selected_array << self[i] if yield(self[i])
+      i += 1
+    end
+    selected_array
+  end
+
+  # my_all
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |el| return false if yield(el) == false }
+      return true
+    elsif arg.nil?
+      my_each { |k| return false if k.nil? || k == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |k| return false if k.class != arg }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |k| return false unless arg.match(k) }
+    else
+      my_each { |k| return false if k != arg }
+    end
+    true
+  end
+
+  #  my_any
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |el| return true if yield(el) }
+      return false
+    end
+    arg.nil? ? arg.class.to_s : my_any? { |el| el }
+
+    if arg.class.to_s == 'Class'
+      my_each { |el| return true if el.is_a? arg }
+    elsif arg.class.to_s == 'Regexp'
+      my_each { |el| return true if el =~ arg }
+    elsif arg.nil?
+      each { |el| return true if el }
+    else
+      my_each { |el| return true if el == arg }
+    end
+    false
+  end
+
+  # my_none
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
```

Comment:
- [ ] `my_none` is not working as the original `none?` method, also is not passing all test cases. 

### `none?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are false.
```
[nil, nil, nil].none?
# true
```
- Returns `true` if all elements are false.
```
(1..3).none?
# false
```

[Enumerable#none?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-none-3F)